### ISSUE TITLE
How to inference on my own point cloud data?

### ISSUE THREAD

**Issue by jackchinor** [filaPro/unidet3d#4]
Nice works , I want to inference on my own point cloud data? what data should I prepare? Thanks.

**Comment by filaPro** [comment]
You need point cloud data in [[x, y, z, r, g, b], ...] format and superpoint clusterinzation of it. Unfortunately this second part is not straightforward and can be done with [scannet segmentator](https://github.com/ScanNet/ScanNet/tree/master/Segmentator) or [superpoint_transformer](https://github.com/drprojects/superpoint_transformer).

**Comment by jackchinor** [comment]
> You need point cloud data in [[x, y, z, r, g, b], ...] format and superpoint clusterinzation of it. Unfortunately this second part is not straightforward and can be done with [scannet segmentator](https://github.com/ScanNet/ScanNet/tree/master/Segmentator) or [superpoint_transformer](https://github.com/drprojects/superpoint_transformer).

thanks. Need any other data? such as .pkl file. Could you give an script example? Thanks very much

**Comment by Nacriema** [comment]
Hi, currently I can create my own ply files that can work with the ScanNet's Segmentator.

**Comment by jackchinor** [comment]
@Nacriema Can you inference on your own ply data?

**Comment by Nacriema** [comment]
Yes, I can run my own ply data with ScanNet's Segmentator. But with the Superpoint transformer I still need more effort to investigate with the 3D Coordinates because I meet the error: 
```
RuntimeError: Trying to create tensor with negative dimension -29: [1369933, -29]
```
when trying to perform segmentation with the original kitti-360 dataset.

**Comment by jackchinor** [comment]
@Nacriema I mean inference your data through unidet3d and output 3d object detection result

**Comment by Nacriema** [comment]
At the moment, the status is no. I am on the way to perform superpoint transformation like the author's suggestion.

**Comment by jackchinor** [comment]
@filaPro Need any other data? such as .pkl file. Could you give an script example or tips? Thanks very much

**Comment by filaPro** [comment]
It is possible to initialize your custom dataset class even without `.pkl` files. They contain mainly just list of paths to point clouds and annotations. But we also provide scripts, how to make them for our datasets in [data](https://github.com/filaPro/unidet3d/tree/master/data) directory.

**Comment by jackchinor** [comment]
> It is possible to initialize your custom dataset class even without `.pkl` files. They contain mainly just list of paths to point clouds and annotations. But we also provide scripts, how to make them for our datasets in [data](https://github.com/filaPro/unidet3d/tree/master/data) directory.

Thanks, but I still don't know how to make a inference on my own point cloud data and super point data, seems it doesn't work when I run tools/test.py . Need your help, thanks
